controller/stats_reporter_test.go

```diff
@@ -0,0 +1,149 @@
+		t.Error("Reporter.Report() expected an error for Report call before init. Got success.")
+	}
+
+	r, _ := NewStatsReporter("testreconciler")
+	wantTags := map[string]string{
+		"reconciler": "testreconciler",
+	}
+
+	// Send statistics only once and observe the results
+	expectSuccess(t, func() error { return r.ReportQueueDepth(10) })
+	checkLastValueData(t, "work_queue_depth", wantTags, 10)
+
+	// Queue depth stats is a gauge - record multiple entries - last one should stick
+	expectSuccess(t, func() error { return r.ReportQueueDepth(1) })
+	expectSuccess(t, func() error { return r.ReportQueueDepth(2) })
+	expectSuccess(t, func() error { return r.ReportQueueDepth(3) })
+	checkLastValueData(t, "work_queue_depth", wantTags, 3)
+}
+
+func TestReportReconcile(t *testing.T) {
+	r, _ := NewStatsReporter("testreconciler")
+	wantTags := map[string]string{
+		"reconciler": "testreconciler",
+		"key":        "test/key",
+		"success":    "true",
+	}
+
+	expectSuccess(t, func() error { return r.ReportReconcile(time.Duration(10), "test/key", "true") })
+	checkCountData(t, "reconcile_count", wantTags, 1)
+	checkDistributionData(t, "reconcile_latency", wantTags, 10)
+
+	expectSuccess(t, func() error { return r.ReportReconcile(time.Duration(15), "test/key", "true") })
+	checkCountData(t, "reconcile_count", wantTags, 2)
+	checkDistributionData(t, "reconcile_latency", wantTags, 25)
+}
+
+func expectSuccess(t *testing.T, f func() error) {
+	if err := f(); err != nil {
+		t.Errorf("Reporter.Report() expected success but got error %v", err)
+	}
+}
+
+func checkLastValueData(t *testing.T, name string, wantTags map[string]string, wantValue float64) {
+	if row := checkRow(t, name); row != nil {
+		checkTags(t, wantTags, row)
+		if s, ok := row.Data.(*view.LastValueData); !ok {
+			t.Error("Reporter.Report() expected a LastValueData type")
+		} else {
+			if s.Value != (float64)(wantValue) {
+				t.Errorf("Reporter.Report() expected %v got %v. metric: %v", s.Value, (float64)(wantValue), name)
+			}
+		}
+	}
+}
+
+func checkCountData(t *testing.T, name string, wantTags map[string]string, wantValue int64) {
+	if row := checkRow(t, name); row != nil {
+		checkTags(t, wantTags, row)
+		if s, ok := row.Data.(*view.CountData); !ok {
+			t.Error("Reporter.Report() expected a LastValueData type")
+		} else {
+			if s.Value != wantValue {
+				t.Errorf("Reporter.Report() expected %v got %v. metric: %v", s.Value, (float64)(wantValue), name)
+			}
+		}
+	}
+}
+
+func checkDistributionData(t *testing.T, name string, wantTags map[string]string, wantValue float64) {
+	if row := checkRow(t, name); row != nil {
+		checkTags(t, wantTags, row)
+		if s, ok := row.Data.(*view.DistributionData); !ok {
+			t.Error("Reporter.Report() expected a LastValueData type")
+		} else {
+			if s.Sum() != wantValue {
+				t.Errorf("Reporter.Report() expected %v got %v. metric: %v", s.Sum(), wantValue, name)
+			}
+		}
+	}
+}
+
+func checkRow(t *testing.T, name string) *view.Row {
```

Comment:
In all of these helpers you want `t.Helper()` so that line numbers are reported more accurately.